src/main/java/io/vertx/core/http/impl/ClientHandler.java

```diff
@@ -107,14 +106,14 @@ protected void handleMessage(ClientConnection conn, ContextImpl context, Channel
         case BINARY:
         case CONTINUATION:
         case TEXT:
+        case PONG:
           conn.handleWsFrame(frame);
           break;
         case PING:
           // Echo back the content of the PING frame as PONG frame as specified in RFC 6455 Section 5.5.2
-          chctx.writeAndFlush(new WebSocketFrameImpl(FrameType.PONG, frame.getBinaryData()));
-          break;
-        case PONG:
-          // Just ignore it
+          WebSocketFrameImpl pong = new WebSocketFrameImpl(FrameType.PONG, frame.getBinaryData());
+          conn.reportBytesWritten(pong.binaryData().length());
```

Comment:
this should be tested

Reply:
The unit tests I wrote in WebSocketTest do validate this functionality.  I reverted these lines and the tests failed.

I looked into writing a lower level test, but the API don't provide access to the underlying connection or clientHandler or `ChannelHandlerContext`.  Similarly, I cannot find a way to do this on the server API.  With enough time I could probably figure out to write an additional low level test. However, these tests would probably be very complex and the time investment/value proposition doesn't seem appropriate since the functionality is already validated by another UnitTest.